src/GameplayKit/GKBehavior.cs

```diff
@@ -18,7 +20,7 @@ public partial class GKBehavior {
 		}
 
 		public NSNumber this [GKGoal goal] {
-			get { return ObjectForKeyedSubscript (goal); }
+			get { return ObjectForKeyedSubscript (goal)!; }
```

Comment:
From here: https://developer.apple.com/documentation/gameplaykit/gkbehavior/1388723-objectforkeyedsubscript?language=objc

`Return Value
The weight to be applied to the goal’s influence on an agent’s speed and direction, or 0.0 if the goal is not in the behavior.`

Reply:
Why do we need the !? I'm looking at the bindings and we have:
```
NSNumber ObjectForKeyedSubscript (GKBehavior behavior);
```

There is no nullable value, is there?

Reply:
@mandel-macaque were you looking at this one inside GKCompositeBehavior https://github.com/xamarin/xamarin-macios/blob/main/src/gameplaykit.cs#L336?
This one refers to the ObjectForKeyedSubscript inside GKBehavior https://github.com/xamarin/xamarin-macios/blob/main/src/gameplaykit.cs#L227 which has a nullable return

Reply:
Note for future TJ: 
The tests show the return value is indeed null when the goal is not in the behavior, add that if this value returns null, throw an ArgumentOutOfRangeException, and file a radar with Apple that the APIs behavior is different than their implementation

Reply:
Note for future TJ: 
The tests show the return value is indeed null when the goal is not in the behavior, add that if this value returns null, throw an ArgumentOutOfRangeException, and file a radar with Apple that the APIs behavior is different than their implementation